transport-native-epoll/src/main/java/io/netty/channel/libaio/DirectFileDescriptor.java

```diff
@@ -0,0 +1,133 @@
+        return from(checkNotNull(file, "file").getPath(), context);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close() throws IOException {
+        open = false;
+        DirectFileDescriptorController.close(fd);
+    }
+
+    /**
+     *
+     * It will submit a write to the queue. The callback sent here will be received on the
+     *   {@link DirectFileDescriptorController#poll(java.nio.ByteBuffer, Object[], int, int).
+
+     * Notice: this won't hold a global reference on buffer, callback should hold a reference towards bufferWrite.
+     *         And don't free the buffer until the callback was called as this could crash the VM.
+     *
+     * @param position The position on the file to write. Notice this has to be a multiple of 512.
+     * @param size The size of the buffer to use while writing.
+     * @param buffer a Native buffer allocated by {@link #newBuffer(int)}.
+     * @param callback A callback to be returned on the poll method.
+     * @throws IOException
+     *
+     * @see DirectFileDescriptorController#poll(ByteBuffer, Object[], int, int)
+     */
+    public void write(long position, int size, ByteBuffer buffer, Object callback) throws IOException {
+        DirectFileDescriptorController.submitWrite(this.fd, libaioContext, position, size, buffer, callback);
+    }
+
+    /**
+     *
+     * It will submit a read to the queue. The callback sent here will be received on the
+     *   {@link DirectFileDescriptorController#poll(java.nio.ByteBuffer, Object[], int, int)}.
+     *
+     * Notice: this won't hold a global reference on buffer, callback should hold a reference towards bufferWrite.
+     *         And don't free the buffer until the callback was called as this could crash the VM.
+     * *
+     * @param position The position on the file to read. Notice this has to be a multiple of 512.
+     * @param size The size of the buffer to use while reading.
+     * @param buffer a Native buffer allocated by {@link #newBuffer(int)}.
+     * @param callback A callback to be returned on the poll method.
+     * @throws IOException
+     *
+     * @see DirectFileDescriptorController#poll(ByteBuffer, Object[], int, int)
+     */
+    public void read(long position, int size, ByteBuffer buffer, Object callback)  throws IOException {
+        DirectFileDescriptorController.submitRead(this.fd, libaioContext, position, size, buffer, callback);
+    }
+
+    /**
+     * It will allocate a buffer to be used on libaio operations.
+     * Buffers here are allocated with posix_memalign.
+     *
+     * You need to explicitly free the buffer created from here using the
+     *          {@link DirectFileDescriptorController#freeBuffer(ByteBuffer)}.
+     *
+     * @param size the size of the buffer.
+     * @return the buffer allocated.
+     */
+    public ByteBuffer newBuffer(int size) {
+        // TODO: do I need to use the alignment of the file itself or not?
```

Comment:
You need to be aligned by 512 on libaio.

However the best really seems to be aligned to the file system alignment.

There is an API (fstat) to get the file's alignment. 
this TODO is to check if that's needed or not.

I think it's fine with 512. but need to double check as it's not clear from the little doc I could find.